javascript/app-discover/src/components/lists/EdgeList.utils.ts

```diff
@@ -0,0 +1,145 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+
+import type { CausalDiscoveryConstraints } from '../../domain/CausalDiscovery/CausalDiscoveryConstraints.js'
+import type { Relationship } from '../../domain/Relationship.js'
+import {
+	hasInvertedSourceAndTarget,
+	hasSameReason,
+	hasSameSourceAndTarget,
+	invertRelationship,
+	ManualRelationshipReason,
+} from '../../domain/Relationship.js'
+
+export function removeEdge(
+	constraints: CausalDiscoveryConstraints,
+	onUpdateConstraints: (newConstraints: CausalDiscoveryConstraints) => void,
+	relationship: Relationship,
+) {
+	const newConstraints = {
+		...constraints,
+		manualRelationships: [
+			...constraints.manualRelationships.filter(
+				r =>
+					!hasSameSourceAndTarget(r, relationship) ||
```

Comment:
Can we express this boolean expression with a named function? `isEquivalentRelationship(...)`?